jmclient/jmclient/wallet.py

```diff
@@ -1013,6 +979,381 @@ def close(self):
     def __del__(self):
         self.close()
 
+class DummyKeyStore(btc.KeyStore):
+    @classmethod
+    def from_iterable(cls, iterable, **kwargs):
+        kstore = cls(**kwargs)
+        for k in iterable:
+            kstore.add_key(k)
+        return kstore
+
+    def add_key(self, k):
+        if isinstance(k, btc.CKeyBase):
```

Comment:
I think it is better to just call `super().add_key(k)`

Reply:
oh; and this is because it will only execute the `isinstance(k, CKeyBase)` part in that function? I see, hmm .. in that case do I need a wrapper class at all? I think I should just create a KeyStore directly with `require_path_templates=False`? That'll be a nice simplification then.

Reply:
Done the above in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/536/commits/9e9ae9e287e2d6e97ea9d53ca628d79fd8e81453

Reply:
`require_path_templates=True` means xkeys must be supplied with path templates. `require_path_templates=False` means xkeys can be supplied with or without path templates.
 
If you just use `KeyStore(require_path_templates=False)` directly, it will accept privkeys, pubkeys, and extended privkeys/pubkeys (both with and without templates)

I had an impression that your DummyKeyStore was about only allowing privkeys (descendants of CKeyBase) to be supplied to add_key()

Reply:
(keystore with pubkeys/xpubs is for use in change detection)

Reply:
Yes, my purpose was only to present an appropriate KeyStore object to the `PartiallySignedTransaction.sign()` function, and creating it from the Joinmarket wallet's keys. I don't think there's a need to enforce a restriction in key type here, as the temporary KeyStore construction is only within the function, it's not exported to callers. Though your further explanation on that point is noted.